src/commands/job/execute.ts

```diff
@@ -102,11 +102,28 @@ export async function execute(
       macroVars?.macroVars
     )
     .catch(async (err) => {
+      if (err?.name === 'NoSessionStatus') {
+        if (err?.logUrl) {
+          const logData = await fetchLogFileContent(
+            sasjs,
+            authConfig.access_token,
+            err.logUrl,
+            100000
+          )
+
+          await saveLog(logData, logFile, jobPath, returnStatusOnly)
+        }
+
+        displayError(err)
+
+        terminateProcess(2)
```

Comment:
Could we not just throw the error here and let the process.exit get called in main.ts?
I think we should only have process.exit calls at the very top level.

Reply:
done